### ISSUE TITLE
Synology Download Station not "keeping" authentication in Medusa

### ISSUE THREAD

**Issue by Sandermand** [pymedusa/Medusa#8810]
I have my Medus install set up on my Synology DS 1315+ NAS.

My authentication from Medusa to my Synology diskstation will not "stick" after a day or so. Medusa will suddenly stop downloading episodes and if i go to the Search Settings and "Test connection" for Torrent search, i get:

![image](https://user-images.githubusercontent.com/27831056/100723513-23aacc00-33c2-11eb-9f24-b93e324c071d.png)

But if i go up to the password field and type in the same password that was alreadyt supposed to be in the field, it works and authenticates:

![image](https://user-images.githubusercontent.com/27831056/100723611-43da8b00-33c2-11eb-9fbb-feae02531904.png)

This happens every couple of days, which means the software is useless, since i have to constantly log in. I have checked the logs but can find no signs in Medusa that the authentication suddenly fails. 

What do ?

![image](https://user-images.githubusercontent.com/27831056/100729706-66bc6d80-33c9-11eb-958b-9a1389e97901.png)

**Comment by p0psicles** [comment]
The setting should be reflected in config.ini.
Please check after configuring it.

**Comment by Sandermand** [comment]
> The setting should be reflected in config.ini.
> Please check after configuring it.

If i look through this config: /volume1/@appstore/medusa/var/config.ini

The only mentions of anything Syno og Downloadstation related is this:

![image](https://user-images.githubusercontent.com/27831056/100742267-cc652580-33da-11eb-8e69-46bcf39dc05c.png)

![image](https://user-images.githubusercontent.com/27831056/100742272-cff8ac80-33da-11eb-87a6-fc1380535c55.png)

None of these seem to have anything to do with authentication.

**Comment by medariox** [comment]
It's saved as:
```
[TORRENT]
torrent_username = ...
torrent_password = ...
```

**Comment by Sandermand** [comment]
> It's saved as:
> 
> ```
> [TORRENT]
> torrent_username = ...
> torrent_password = ...
> ```

Found em, changed the password from the current encrypted password to the "plaintext correct password", restarted Medusa, it "reencrypted" the password in the config.ini again and still not working. I can see the same encrypted password is the same as before, so it is the same saved password.

Still not authenticating without me manually typing in the password in the field.

**Comment by p0psicles** [comment]
Can't you use black hole for sending the torrents?

**Comment by Sandermand** [comment]
> Can't you use black hole for sending the torrents?
> Have you tried with setting "encryping passwords" to off?

Dont know how to use Black Hole.

Unchecking "Encrypt Passwords" revealed the issue, i think:

Medusa hashes the Password "wrong". 

If i uncheck that setting and restart the Medusa service, the config.ini now contains the "original" password as Medusa thinks it looks like by un-hashing it. And it is a jumbled mess of some of the correct characters and numbers.

Fx. (not my password, just to make my point):

Correct DS password, typed into "Search Settings":
**Malcontent5043**

Hashed password in config.ini with "Encrypt password" on:
**fRleUgxZRQRXRwRRVQA=**

If i now restart the server and check the config.ini, this is the hashed password (notice it is not the same as above, the hash changed without me doing anything):
**fRlYVV9bRFNWRQVWUgM=**

If i toggle Encrypt off, this is the clear text password in the Config.ini, extrapolated from the "wrong" hash in my last step above:
**Mando8&kguf3dg**

So, something happens in regular intervals and during restart, that messes up the hash of the saved password for torrent Search Settings. Maybe some of the parameters for cipher is borked during runtime or at scheduled intervals ?

I dont want to leave my passwords for all my trackers unencrypted in the config files, so what is there to do ?

**Comment by p0psicles** [comment]
I guess wait until someone fixes it

**Comment by p0psicles** [comment]
You can also use black hole. It's one of the options in the select drop down with all the clients. Basically you set it to a folder, where it will  drop the torrent in. Then you configure download station to pick it up from that folder.

**Comment by p0psicles** [comment]
Unfortunately I can't reproduce the issue.

**Comment by p0psicles** [comment]
Can you check your config.ini for the encryption_secret value? How is it formatted?

**Comment by p0psicles** [comment]
I created a branch, which might have a fix. But you'll have to test it out.
`git fetch origin`
`git checkout -t origin/feature/fix-encrypt-decrypt`

Then you have to go into your config.ini.
And delete the lines
web_cookie_secret = ...
encryption_secret = ...

Now start medusa. It should create new values. You might need to reset some passwords. But they should stay saved now.

**Comment by Sandermand** [comment]
> Can you check your config.ini for the encryption_secret value? How is it formatted?

Encryption Secret is like this:
![image](https://user-images.githubusercontent.com/27831056/101017034-d8cbb880-3569-11eb-82e1-9bb0e78018d7.png)

**Comment by Sandermand** [comment]
> I created a branch, which might have a fix. But you'll have to test it out.
> `git fetch origin`
> `git checkout -t origin/feature/fix-encrypt-decrypt`
> 
> Then you have to go into your config.ini.
> And delete the lines
> web_cookie_secret = ...
> encryption_secret = ...
> 
> Now start medusa. It should create new values. You might need to reset some passwords. But they should stay saved now.

How do i fire off that command ? I am no linux pro :) Get this error in putty:

![image](https://user-images.githubusercontent.com/27831056/101017366-48da3e80-356a-11eb-85ba-0e4fe74fa780.png)

**Comment by p0psicles** [comment]
Run it in the medusa program folder

**Comment by Sandermand** [comment]
> Run it in the medusa program folder

No dice:
![image](https://user-images.githubusercontent.com/27831056/101018984-9061ca00-356c-11eb-8915-c4af9e4f6b49.png)

**Comment by p0psicles** [comment]
Ok never mind

**Comment by Sandermand** [comment]
I'll use BlackHole for now until something comes from this bug report :)

**Comment by p0psicles** [comment]
Should be fixed in develop branch. You could try to check that out using UI. Else wait til next release

**Comment by FransAlmere** [comment]
Same problem here with using Synology its downloadstation for the use of torrents. When testing its oke but get the message 

![Medusa Synology Torrent](https://user-images.githubusercontent.com/13037978/101362111-f9fd1380-389f-11eb-8ebe-38b3ef317213.jpg)

If I test the settings in Medusa all is oke

If I use the ip and user and password in the web I can login

It did not help switching to develop branch

and also removing web.cookie etc.

Can it have something to do wirh the ssl certificate ?

Password and user info are not encrypted in config.ini

I wait to see if the problem can be solved.

Thanks on beforehand

**Comment by medariox** [comment]
@FransAlmere 
Check this for the fix: https://github.com/pymedusa/Medusa/issues/8822
Just remove /volume1 from the path.

**Comment by FransAlmere** [comment]
Gonna try thuis but I got it also Working with Black Hole. 


Thank for the Quick response 


Frans Lieshout

Banjostraat 49

1312 KB Almere 


Deze mail is verzonden van mijn Motorola via Outlook. 


De informatie in deze e-mail is vertrouwelijk en uitsluitend bestemd voor de geadresseerde. Wanneer u dit bericht per abuis ontvangt, verzoekt Frans Lieshout u contact op te nemen met de afzender. Bovendien verzoekt Frans Lieshout u in dat geval deze e-mail te vernietigen en de inhoud aan niemand openbaar te maken. Frans Lieshout aanvaardt geen aansprakelijkheid voor onjuiste, onvolledige dan wel ontijdige overbrenging van de inhoud van een verzonden e-mail, noch voor overgebrachte virussen.

**Comment by Sandermand** [comment]
> Same problem here with using Synology its downloadstation for the use of torrents. When testing its oke but get the message
> 
> ![Medusa Synology Torrent](https://user-images.githubusercontent.com/13037978/101362111-f9fd1380-389f-11eb-8ebe-38b3ef317213.jpg)
> 
> If I test the settings in Medusa all is oke
> 
> If I use the ip and user and password in the web I can login
> 
> It did not help switching to develop branch
> 
> and also removing web.cookie etc.
> 
> Can it have something to do wirh the ssl certificate ?
> 
> Password and user info are not encrypted in config.ini
> 
> I wait to see if the problem can be solved.
> 
> Thanks on beforehand

Yeah, that is NOT the same issue that i am having...

**Comment by FransAlmere** [comment]
Hi, Sandermand

 

Dario, allready answered my question I deleted in the downloaded files location /volume../

 



 

And its working now I also now can use Black Hole

 



 

Tested this and for me its now working

Frans Lieshout

Banjostraat 49
1312KB Almere
 
Tel: +31(36)7441527
Mobiel: +31(6)22406833

Email: info@franslieshout.nl
Web : https://www.franslieshout.nl




De informatie in deze e-mail is vertrouwelijk en uitsluitend bestemd voor de geadresseerde. Wanneer u dit bericht per abuis ontvangt, verzoekt Frans Lieshout u contact op te nemen met de afzender. Bovendien verzoekt Frans Lieshout u in dat geval deze e-mail te vernietigen en de inhoud aan niemand openbaar te maken. Frans Lieshout aanvaardt geen aansprakelijkheid voor onjuiste, onvolledige dan wel ontijdige overbrenging van de inhoud van een verzonden e-mail, noch voor overgebrachte virussen.

 

Van: Sandermand <notifications@github.com> 
Verzonden: dinsdag 8 december 2020 09:17
Aan: pymedusa/Medusa <Medusa@noreply.github.com>
CC: FransAlmere <info@franslieshout.nl>; Mention <mention@noreply.github.com>
Onderwerp: Re: [pymedusa/Medusa] Synology Download Station not "keeping" authentication in Medusa (#8810)

 

Same problem here with using Synology its downloadstation for the use of torrents. When testing its oke but get the message

 <https://user-images.githubusercontent.com/13037978/101362111-f9fd1380-389f-11eb-8ebe-38b3ef317213.jpg> 

If I test the settings in Medusa all is oke

If I use the ip and user and password in the web I can login

It did not help switching to develop branch

and also removing web.cookie etc.

Can it have something to do wirh the ssl certificate ?

Password and user info are not encrypted in config.ini

I wait to see if the problem can be solved.

Thanks on beforehand

Yeah, that is NOT the same issue that i am having...
